x-pack/plugins/security_solution/public/detection_engine/rule_response_actions/endpoint/field_name.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useEffect } from 'react';
+import {
+  getFieldValidityAndErrorMessage,
+  UseField,
+  useFormData,
+  useFormContext,
+} from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { get } from 'lodash';
+import { EuiComboBox, EuiSpacer, EuiFormRow } from '@elastic/eui';
+import ECSSchema from './v.8.10.0_process.json';
+
+interface FieldNameFieldProps {
+  path: string;
+  disabled: boolean;
+  readDefaultValueOnForm: boolean;
+  isRequired: boolean;
+}
+
+const ECSSchemaOptions = ECSSchema.map((ecs) => ({
+  label: ecs.field,
+  value: ecs,
+}));
+
+const SINGLE_SELECTION = { asPlainText: true };
+const FIELD_LABEL: string = 'Custom field name';
+const FieldNameFieldComponent = ({
+  path,
+  disabled,
+  readDefaultValueOnForm,
+  isRequired,
+}: FieldNameFieldProps) => {
+  const [data] = useFormData();
+  const fieldValue = get(data, path);
+  const context = useFormContext();
+
+  const currentFieldNameField = context.getFields()[path];
+
+  useEffect(() => {
+    // hackish way to clear errors on this field - because we base this validation on the value of overwrite toggle
+    if (currentFieldNameField && !isRequired) {
+      currentFieldNameField?.clearErrors();
+    }
+  }, [currentFieldNameField, isRequired]);
+
+  const renderEntityIdNote = useMemo(() => {
+    const contains = fieldValue?.includes('entity_id');
+    if (contains) {
+      return (
+        <FormattedMessage
+          id="xpack.securitySolution.responseActions.endpoint.fieldDescription"
+          defaultMessage="Entity_id is an endpoint specific field, if the alert does not come from endpoint we will not be able to send the action."
```

Comment:
Instead of `endpoint` please use `Elastic Security agent` instead? (the word `endpoint` is used often for other types of "endpoint")